### ISSUE TITLE
Invalid model for QF_NIA instances  (non-incremental and no options about rewriter)

### ISSUE THREAD

**Issue by rainoftime** [Z3Prover/z3#3379]
Hi, for the following max-smt instances,
[invalid.txt](https://github.com/Z3Prover/z3/files/4342061/invalid.txt)
[invalid2.txt](https://github.com/Z3Prover/z3/files/4342078/invalid2.txt)

z3 master branch (Commit 1c5283f3a44) gives an invalid model
~~~~
(error "line 51 column 10: an invalid model was generated")
~~~~

The option is 
~~~~
(set-option :model_validate true)
~~~~

**Comment by rainoftime** [comment]
Debug branch aa119d12e35d8f also has the issue

**Comment by rainoftime** [comment]
@NikolajBjorner The root causes might be the same with https://github.com/Z3Prover/z3/issues/3244 (about incremental solving).  The maxres algorithm uses the incremental solver inside `smt` tactic.

**Comment by rainoftime** [comment]
This issue has been fixed in the master branch 752b498254ba.
Perhaps debug branch needs to patch.
